cmd/otel-allocator/prehooktargetfilter/relabel_config_filter.go

```diff
@@ -0,0 +1,101 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package prehooktargetfilter
+
+import (
+	"fmt"
+	"strings"
+	"sync"
+
+	"github.com/go-logr/logr"
+	"github.com/prometheus/common/model"
+	"github.com/prometheus/prometheus/model/relabel"
+
+	"github.com/open-telemetry/opentelemetry-operator/cmd/otel-allocator/allocation"
+)
+
+type RelabelConfigTargetFilter struct {
+	m           sync.RWMutex
+	log         logr.Logger
+	targetItems map[string]*allocation.TargetItem
+	allocator   allocation.Allocator
+}
+
+func NewRelabelConfigTargetFilter(log logr.Logger, allocator allocation.Allocator) AllocatorPrehook {
+	return &RelabelConfigTargetFilter{
+		log:         log,
+		allocator:   allocator,
+		targetItems: make(map[string]*allocation.TargetItem),
+	}
+}
+
+func (tf *RelabelConfigTargetFilter) SetTargets(targets map[string]*allocation.TargetItem) {
+	numRemainingTargets := 0
+	for jobName, tItem := range targets {
+		keepTarget := true
+		for _, cfg := range tItem.RelabelConfigs {
+			if tf.IsDropTarget(tItem.Label, cfg) {
+				keepTarget = false
+				break
+			}
+		}
+
+		if keepTarget {
+			tf.targetItems[jobName] = tItem
+			numRemainingTargets++
+		}
+	}
+
+	tf.allocator.SetTargets(tf.targetItems)
+	tf.log.Info(fmt.Sprintf("Relabel filtering completed. Keeping %d target(s) out of %d total targets", numRemainingTargets, len(targets)))
+}
+
+// TargetItems returns a shallow copy of the targetItems map.
+func (tf *RelabelConfigTargetFilter) TargetItems() map[string]*allocation.TargetItem {
```

Comment:
Can this just defer to the wrapped allocator?  I'm not sure this behavior is correct.

Reply:
Good point I have now replaced this function with ```	return tf.allocator.TargetItems()```